File: src/main/typescript/wcardinal/controller/page-factory.ts
/*
 * Copyright (C) 2019 Toshiba Corporation
 * SPDX-License-Identifier: Apache-2.0
 */

import { Factory } from "./factory";
import { Page } from "./page";

/**
 * A page factory that creates/destroys components dynamically.
 */
export interface PageFactory extends Factory<Page> {}

File: src/main/java/org/wcardinal/controller/PageFactory.java
/*
 * Copyright (C) 2019 Toshiba Corporation
 * SPDX-License-Identifier: Apache-2.0
 */

package org.wcardinal.controller;

/**
 * A factory that creates/destroys the specified class {@code T} as a page dynamically.
 *
 * @param <T> page class to be created/destroyed dynamically
 */
public interface PageFactory<T> extends Factory<T> {

}
